support-frontend/app/services/stepfunctions/SupportWorkersClient.scala

```diff
@@ -128,7 +128,7 @@ class SupportWorkersClient(
       giftRecipient = getGiftRecipient(request.body),
       None, // TODO
       product = request.body.product,
-      paymentFields = request.body.paymentFields,
+      paymentFields = request.body.paymentFields.map(PaidProduct(_)).getOrElse(FreeProduct),
```

Comment:
I think we might have already discussed this, but since the types can't ensure that the sub is a corporate digital pack at this stage, we need a clear validation at this point.  i could see some existing separate validation but I think it might need a new validator.
The existing validator has actually suffered the "option" issue again in that we only validate that the individual fields are present if the whole are present, but not the conditions that the whole should be present https://github.com/guardian/support-frontend/pull/2484/files#diff-a7bab7639dad7567439a537a17cfc4a8R22

Reply:
Have refactored the validators to take account of the new type. There is still some work to do here in phase 2 but not for this PR